app/src/main/java/com/mk/tasky/authentication/login/data/LoginRepositoryImpl.kt

```diff
@@ -0,0 +1,21 @@
+package com.mk.tasky.authentication.login.data
+
+import com.mk.tasky.authentication.login.data.mapper.toDomain
+import com.mk.tasky.authentication.login.data.remote.LoginApi
+import com.mk.tasky.authentication.login.data.remote.dto.LoginBodyDto
+import com.mk.tasky.authentication.login.domain.LoginRepository
+import com.mk.tasky.authentication.login.domain.models.LoggedUser
+
+class LoginRepositoryImpl(
+    private val api: LoginApi
+) : LoginRepository {
```

Comment:
Naming suggestion: `AuthRepository` since I assume you don't want a separate repo for every single API route, right?

Reply:
Should I then move the Data and Domain packages, inside the Authentication package instead of the Login package?
So it can be later used in the Registration package
Or should I create a core package with all the Data and Domain and have all features use it?

Reply:
Personally, I did it that way for this project. I have an AuthApi interface which contains functions for login, registration, checking auth status and logging out. I don't really see a point in having a separate repo for all these functions. You also don't **need** to use a repo for this, you could also call it something like `TaskyAuthenticator` or so.